src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -0,0 +1,371 @@
+                }
+                yield l;
+            }
+            case RuntimeInvisibleAnnotationsAttribute aa -> {
+                int l = 2;
+                for (var an : aa.annotations()) {
+                    l += annotationSize(an);
+                }
+                yield l;
+            }
+            case RuntimeVisibleParameterAnnotationsAttribute aa -> {
+                int l = 1;
+                for (var ans : aa.parameterAnnotations()) {
+                    l += 2;
+                    for (var an : ans) {
+                        l += annotationSize(an);
+                    }
+                }
+                yield l;
+            }
+            case RuntimeInvisibleParameterAnnotationsAttribute aa -> {
+                int l = 1;
+                for (var ans : aa.parameterAnnotations()) {
+                    l += 2;
+                    for (var an : ans) {
+                        l += annotationSize(an);
+                    }
+                }
+                yield l;
+            }
+            case SignatureAttribute sa -> 2;
+            case SourceDebugExtensionAttribute sda -> sda.contents().length;
+            case SourceFileAttribute sfa -> 2;
+            case SourceIDAttribute sida -> 2;
+            case SyntheticAttribute sa -> 0;
+            default -> payLoad;
+        }) {
+            errors.add(new VerifyError("Wrong %s attribute length in %s".formatted(a.attributeName(), toString(ae))));
+        }
+
+    }
+
+    private static int annotationSize(Annotation an) {
+        int l = 4;
+        for (var el : an.elements()) {
+            l += 2 + valueSize(el.value());
+        }
+        return l;
+    }
+
+    private static int valueSize(AnnotationValue val) {
+        return 1 + switch (val) {
+            case AnnotationValue.OfAnnotation oan ->
+                annotationSize(oan.annotation());
+            case AnnotationValue.OfArray oar -> {
+                int l = 2;
+                for (var v : oar.values()) {
+                    l += valueSize(v);
+                }
+                yield l;
+            }
+            case AnnotationValue.OfConstant _, AnnotationValue.OfClass _ -> 2;
+            case AnnotationValue.OfEnum _ -> 4;
+        };
+    }
+
+    private String className() {
+        return classModel.thisClass().asSymbol().displayName();
+    }
+
+    private String toString(AttributedElement ae) {
+        return switch (ae) {
+            case CodeModel m -> "Code attribute for " + toString(m.parent().get());
+            case FieldModel m -> "field %s.%s".formatted(
+                    className(),
+                    m.fieldName().stringValue());
+            case MethodModel m -> "method %s::%s(%s)".formatted(
+                    className(),
+                    m.methodName().stringValue(),
+                    m.methodTypeSymbol().parameterList().stream().map(ClassDesc::displayName).collect(Collectors.joining(",")));
```

Comment:
```suggestion
                    m.methodTypeSymbol().displayDescriptor();
```
Can remove the parentheses above.

Reply:
It intentionally prints only parameters and avoids return type.